makefiles/arch/riscv.inc.mk

```
@@ -30,8 +30,15 @@ TARGET_ARCH_RISCV ?= \
 TARGET_ARCH ?= $(TARGET_ARCH_RISCV)
 
 # define build specific options
-CFLAGS_CPU   = -march=rv32imac -mabi=ilp32 -mcmodel=medlow -msmall-data-limit=8
-CFLAGS_LINK  = -nostartfiles -ffunction-sections -fdata-sections
+CFLAGS_CPU   = -march=rv32imac -mabi=ilp32
+ifeq ($(TOOLCHAIN),llvm)
+  # Always use riscv32-none-elf as target triple for clang, as some
+  # autodetected gcc target triples are incompatible with clang
+  TARGET_ARCH := riscv32-none-elf
```

```suggestion
  TARGET_ARCH_LLVM := riscv32-none-elf
```

This and rebase should fix compilation. But clang11 or `FEATURES_REQUIRED=newlib` is needed to get it working now. (Another regression due to changes in upstream.)